task/sast-coverity-check/0.1/sast-coverity-check.yaml

```diff
@@ -0,0 +1,243 @@
+        optional: false
+    - name: auth-token-coverity-image
+      secret:
+        secretName: $(params.AUTH_TOKEN_COVERITY_IMAGE)
+        optional: false
+  steps:
+    - name: sast-coverity-check
+      onError: stopAndFail
+    # TODO: Change image
+      #image: $(steps.secrets-check.results.image)
+      image: quay.io/redhat-user-workloads/sast-tenant/sast-scanner/coverity@sha256:1e6d26ef45da80128c6bbb4f29a29ac3e547e934e533cc5c1dd37e40439269fe
+      computeResources:
+        requests:
+          memory: "16Gi"
+          cpu: "8"
+        limits:
+          memory: "32Gi"
+          cpu: "16"
+      # per https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: cov-license
+          mountPath: "/etc/secrets/cov"
+          readOnly: true
+      env:
+        - name: COV_ANALYZE_ARGS
+          value: $(params.COV_ANALYZE_ARGS)
+        - name: COV_CAPTURE_ARGS
+          value: $(params.COV_CAPTURE_ARGS)
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: COV_LICENSE
+          value: $(params.COV_LICENSE)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+        - name: PROJECT_NVR
+          value: $(params.PROJECT_NVR)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+      script: |
+        #!/usr/bin/env bash
+        set -eo pipefail
+        . /usr/local/share/konflux-test/utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+        
+        echo 'Starting Coverity buildless scan'
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)
+        COVERITY_DIR=/tmp/cov-scan/cov
+        COVERITY_RESULTS_FILE=coverity-buildless-results.js
+        COV_LICENSE_PATH=/etc/secrets/cov/cov-license
+        # Installing Coverity license
+        cp "$COV_LICENSE_PATH" /opt/coverity/bin/license.dat
+        if test -z "$(find /opt -name 'license.dat')"; then
+          echo "No license file for Coverity was detected. Exiting..."
+          exit 1
+        fi
+        
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        curl -sS https://certs.corp.redhat.com/certs/2015-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2015-RH-IT-Root-CA.crt
+        curl -sS https://certs.corp.redhat.com/certs/2022-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2022-IT-Root-CA.pem
+        update-ca-trust
+        
+        # Create configuration file for coverity buildless
+        echo -e 'capture:\n   build-command-inference: false' > "$SOURCE_CODE_DIR"/coverity.yml
+        
+        # Captureless scan
+        env COV_HOST=konflux HOME=/tmp/cov-scan/home /opt/coverity/bin/coverity capture $COV_CAPTURE_ARGS --project-dir $SOURCE_CODE_DIR --dir $COVERITY_DIR
+        COV_CAPTURE_EXIT_CODE=$?
```

Comment:
If this was suggested by me, I was wrong because I did not realize that the whole shell script is executed with `set -e`.  We either need to disable `set -e` temporarily or use the ugly `COV_CAPTURE_EXIT_CODE=0` and `... || COV_CAPTURE_EXIT_CODE=$?` construction.

Reply:
Temporarily disabled `-e`

Reply:
Sure! Added